### DISCUSSION TITLE
How to use the javascript node in the automation part ?

### DISCUSSION THREAD

**Discussion by oswaldB** [Budibase/budibase#1737] [Help] [answered]
Hello,

I would like to know if anyone has ever used this node and can show us how to use it.

My questions
+ how to return a 'hello world' (return 'Hello world'; ?)
+ can I use console.log ?

thanks a lot

**Comment by mike12345567** [comment, marked as answer]
Hi @oswaldB - code ran in the Javascript automation step is ran serverside, in a virtual machine context, generated by the `vm` node module - that can be seen here: https://nodejs.org/api/vm.html

This means that if you want to "return hello world" you would do something like `var x = "hello world"` and x would now be available in the automation context to use in future steps. This is more a transformer, you can't use other node modules etc, right now it is for taking data from one place and transforming it to use for another step.

**Reply by oswaldB** [reply]
Thanks a lot.

Can I use {{ trigger.row }} as well?

**Reply by mike12345567** [reply]
You can yes, for example you could write, `let output = trigger.row.firstName` and then use the binding `{{ output }}` in future automation steps.

**Comment by oswaldB** [comment]
Perfect thank you ;)

On Fri, 18 Jun 2021 at 18:05, Michael Drury ***@***.***>
wrote:

> You can yes, for example you could write, let output =
> trigger.row.firstName and then use the binding {{ output }} in future
> automation steps.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Budibase/budibase/discussions/1737#discussioncomment-889376>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAXCFJTESTLS5ZRUV5P7KU3TTN4C3ANCNFSM465ZDHXA>
> .
>


-- 
Oswald BERNARD
00 44 7861004200

**Comment by Lsm-CVX** [comment]
Hi all, 

I am trying to use the JS scripting for my application and I am a bit confused on the way to make it works. To describe it a bit : I need to have the automatic ID line converted to hex numbers for compatibility reasons 

The automation is : 
 When this happens...
Trigger
Row is added to Tests

Do this...
Step 1
Execute JavaScript Code

Do this...
Step 2
Update a Tests row


My JS code in the automate section is : 

function decimalToHex(d, padding) {
    var hex = Number(d).toString(16);
    padding = typeof (padding) === "undefined" || padding === null ? padding = 5 : padding;

    while (hex.length < padding) {
        hex = "0" + hex;
    }

    return hex;
}
let output = decimalToHexString(trigger.id,5);

The output variable is used as {{ output }} int the TAG line where it is supposed to be updated

I do miss something but I am not able to make it work even without the JS so ...

Thanks in advance, 
Laurent

**Reply by Lsm-CVX** [reply]
Okay I think that I have understood some of the issue : the Id is not the value of the row Id and not the Id of the update itself...

Do you know how can we access the value of one of the row directly in the JS code ?
